clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -244,6 +244,8 @@ class CIRGenBuilderTy : public mlir::OpBuilder {
     }
 
     if (const auto arrayVal = attr.dyn_cast<mlir::cir::ConstArrayAttr>()) {
+      if (arrayVal.getElts().isa<mlir::StringAttr>())
```

Comment:
While here, what should happen for the following cases below?
- There are no elements: `""`.
- The string is null terminated `"\0"`.

Reply:
Ideally, I think this should return true if all characters in the string are `\0`.

Reply:
Yes, for zero strings this branch is true as well. I've added tests to check suggested cases.